python_src/tests/ingestion/test_gtfs_converter.py

```diff
@@ -300,17 +303,92 @@ def test_schedule_conversion() -> None:
         },
     }
 
-    for prefix, table in zip_to_pyarrow(gtfs_schedule_file):
-        table_name = prefix.lower()
 
-        # check that we are expecting this name
-        assert table_name in field_names.keys()  # pylint: disable=C0201
+@pytest.fixture(autouse=True, name="table_check_patch")
+def fixture_table_check(monkeypatch: MonkeyPatch) -> Iterator[None]:
+    """
+    insert a monkeypatch over the s3 write parquet file function.
+    """
+
+    def mock_write_parquet_file(
+        table: Table,
+        config_type: str,
+        s3_path: str,
+        partition_cols: List[str],
+        visitor_func: Callable[..., None],
+    ) -> None:
+        """
+        instead of writing the parquet file to s3, inspect the contents of the
+        table instead
+        """
+        # pull the name out of the s3 path and check that we are expecting this table
+        table_name = config_type.lower()
+        table_attributes = all_table_attributes()[table_name]
 
         # check that this table has all of the field names we're expecting
-        for field in field_names[table_name]["column_names"]:  # type: ignore
+        for field in table_attributes["column_names"]:  # type: ignore
             assert field in table.column_names
 
-        # check that date information is in this table
+        # check that date information is in this table and that we're partitioning it
         assert "timestamp" in table.column_names
+        assert partition_cols == ["timestamp"]
+
+        assert table_attributes["length"] == table.num_rows
+
+        # call the visitor function, which should add the string to the queue to check later
+        visitor_func(os.path.join(s3_path, "written.parquet"))
+        return
+
+    monkeypatch.setattr(
+        "lamp_py.ingestion.convert_gtfs.write_parquet_file",
+        mock_write_parquet_file,
+    )
+
+    def mock_move_s3_objects(files: List[str], to_bucket: str) -> List[str]:
+        if len(files) > 0:
+            assert "archive" in to_bucket
+        return []
+
+    monkeypatch.setattr(
+        "lamp_py.ingestion.convert_gtfs.move_s3_objects", mock_move_s3_objects
+    )
+
+    yield
+
+
+def test_schedule_conversion() -> None:
```

Comment:
i think this reads pretty straight forwardly. i added comments where i think that helps.